.ci/containers/terraform-tester/test_terraform.sh

```diff
@@ -25,112 +17,25 @@ else
-    exit 0
-else
-    echo "Running tests: Go files changed"
-fi
-
-post_body=$( jq -n \
-    --arg context "${gh_repo}-test" \
-    --arg target_url "https://console.cloud.google.com/cloud-build/builds;region=global/${build_id};step=${build_step}?project=${project_id}" \
-    --arg state "pending" \
-    '{context: $context, target_url: $target_url, state: $state}')
-
-curl \
-  -X POST \
-  -u "$github_username:$GITHUB_TOKEN" \
-  -H "Accept: application/vnd.github.v3+json" \
-  "https://api.github.com/repos/$repo_name/statuses/$mm_commit_sha" \
-  -d "$post_body"
-
-
-post_body=$( jq -n \
-    --arg context "${gh_repo}-lint" \
-    --arg target_url "https://console.cloud.google.com/cloud-build/builds;region=global/${build_id};step=${build_step}?project=${project_id}" \
-    --arg state "pending" \
-    '{context: $context, target_url: $target_url, state: $state}')
-
-curl \
-  -X POST \
-  -u "$github_username:$GITHUB_TOKEN" \
-  -H "Accept: application/vnd.github.v3+json" \
-  "https://api.github.com/repos/$repo_name/statuses/$mm_commit_sha" \
-  -d "$post_body"
-
-set +e
-
-make
-lint_exit_code=$?
-test_exit_code=1
-
-# TODO: remove on 2022/08/18 or later.
-# note: maintain (degraded) backwards compatibility with the old lint rule which
-# needs tools run in advance
-# note: this will silently fail on newer makefiles, as we are in a +e block
-make tools
-
-make lint
-lint_exit_code=$(($lint_exit_code || $?))
-
-# note: test has a dependency on lint (which runs fmtcheck and vet) so it will
-# run them a second time.
-make test
-test_exit_code=$?
-
-
-make docscheck
-lint_exit_code=$(($lint_exit_code || $?))
-
-set -e
-
-if [ $test_exit_code -ne 0 ]; then
-    test_state="failure"
-else
-    test_state="success"
-fi
-
-if [ $lint_exit_code -ne 0 ]; then
-    lint_state="failure"
-else
-    lint_state="success"
-fi
-
-post_body=$( jq -n \
-    --arg context "${gh_repo}-test" \
-    --arg target_url "https://console.cloud.google.com/cloud-build/builds;region=global/${build_id};step=${build_step}?project=${project_id}" \
-    --arg state "${test_state}" \
-    '{context: $context, target_url: $target_url, state: $state}')
-
-curl \
-  -X POST \
-  -u "$github_username:$GITHUB_TOKEN" \
-  -H "Accept: application/vnd.github.v3+json" \
-  "https://api.github.com/repos/$repo_name/statuses/$mm_commit_sha" \
-  -d "$post_body"
-
 
-post_body=$( jq -n \
-    --arg context "${gh_repo}-lint" \
-    --arg target_url "https://console.cloud.google.com/cloud-build/builds;region=global/${build_id};step=${build_step}?project=${project_id}" \
-    --arg state "${lint_state}" \
-    '{context: $context, target_url: $target_url, state: $state}')
+post_body=$(jq -n \
+  --arg owner "$github_username" \
+  --arg branch "$new_branch" \
+  --arg repo "$gh_repo" \
+  --arg sha "$mm_commit_sha" \
+  '{
+    ref: "main",
+    inputs: {
+      owner: $owner,
+      repo: $repo,
+      branch: $branch,
+      sha: $sha
+    }
+  }')
 
 curl \
```

Comment:
I think we may need to make sure that if this call fails, cloudbuild fails? Might already be the case.

Reply:
whoops, this is better covered in the other thread.

Reply:
I actually think this a valid concern. `CURL` won't have a bad exit code if server returns a 400. BUT this will fail the entire build pipeline if I exit 1 here. 

IF there is a failure in the curl its a problem with the github api. I'm not sure it's worth creating a mechanism for this kind of rare/unexpected occurrence.